chainer/function_node.py

```diff
@@ -0,0 +1,591 @@
+        :meth:`backward`.
+
+        .. note::
+
+           It is recommended to use this method if the function requires some
+           or all output arrays in backprop. The function can also use output
+           arrays just by keeping references to them directly, whereas it might
+           influence on the performance of later function applications to the
+           output variables.
+
+        Note that **this method must not be called from the outside of
+        forward method.**
+
+        Args:
+            indexes (iterable of int): Indexes of input variables that the
+                function does not require for backprop.
+
+        """
+        self._output_indexes_to_retain = indexes
+
+    def backward(self, target_input_indexes, grad_outputs):
+        """Computes gradients w.r.t. specified inputs given output gradients.
+
+        This method is used to compute one step of the backpropagation
+        corresponding to the forward computation of this function node.
+        Given the gradients w.r.t. output variables, this method computes the
+        gradients w.r.t. specified input variables. Note that this method does
+        not need to compute any input gradients not specified by
+        ``target_input_indices``.
+
+        Unlike :meth:`Function.backward`, gradients are given as
+        :class:`Variable` objects and this method itself has to return input
+        gradients as :class:`Variable` objects. It enables the function node to
+        return the input gradients with the full computational history, in
+        which case it supports *differentiable backpropagation* or
+        *higher-order differentiation*.
+
+        The default implementation returns ``None`` s, which means the
+        function is not differentiable.
+
+        Args:
+            target_input_indexes (tuple of int): Indices of the input variables
+                w.r.t. which the gradients are required. It is guaranteed that
+                this tuple contains at least one element.
+            grad_outputs (tuple of Variable): Gradients w.r.t. the output
+                variables. If the gradient w.r.t. an output variable is not
+                given, the corresponding element is ``None``.
+
+        Returns:
+            Tuple of variables that represent the gradients w.r.t. specified
+            input variables. The length of the tuple can be same as either
+            ``len(target_input_indexes)`` or the number of inputs. In the
+            latter case, the elements not specified by ``target_input_indexes``
+            will be discarded.
+
+        .. seealso::
+
+           :meth:`backward_accumulate` provides an alternative interface that
+           allows you to implement the backward computation fused with the
+           gradient accumulation.
+
+        """
+        return (None,) * len(target_input_indexes)
+
+    def backward_accumulate(self, target_input_indexes, grad_outputs,
```

Comment:
Ah, yes. I didn't originally think of EmbedID, but this interface is basically intended to be used for the cases that accumulation can be made efficient by fusing the operations with backward itself. EmbedID should be one of such cases.